protocols/relay/src/behaviour/handler.rs

```diff
@@ -660,11 +668,63 @@ impl ConnectionHandler for Handler {
         }
 
         // Process protocol requests
-        if let Poll::Ready(worker_res) = self.protocol_futs.poll_unpin(cx) {
-            let event = worker_res
-                .unwrap_or_else(|_| ConnectionHandlerEvent::Close(StreamUpgradeError::Timeout));
-
-            return Poll::Ready(event);
+        match self.workers.poll_unpin(cx) {
+            Poll::Ready(Ok(Either::Left(Ok(Either::Left(inbound_reservation_req))))) => {
```

Comment:
@thomaseizinger
`Poll::Ready(Ok(Either::Left(Ok(Either::Left(inbound_reservation_req)))))`
IMHO, It is hard to keep in mind what means every Either::Left.
Maybe it would be more convenient to create a certain `enum` for every `Either`?

Reply:
Yeah we could! To be honest, I mostly rely on the compiler here to make sure I don't miss any combinations. Personally, I am not too worried about the complexity of these matches because it is very hard to make a mistake :)

If you feel strongly, feel free to make a custom enum but I am not sure it is worth the code addition.

Reply:
>  I am not sure it is worth the code addition.
ok